sdjournal/read.go

```diff
@@ -188,6 +190,83 @@ process:
 	return
 }
 
+func (r *JournalReader) FollowWithFields(until <-chan time.Time, fields map[string]struct{}, writer io.Writer) (err error) {
+	enc := json.NewEncoder(writer)
+journal:
+	for {
+		kvMap := make(map[string]string)
+		select {
+		case <-until:
+			return ErrExpired
+		default:
+			var err error
+			var c int
+
+			// Advance the journal cursor
+			c, err = r.journal.Next()
+
+			// An unexpected error
+			if err != nil {
+				return err
+			}
+
+			// We have a new journal entry go over the fields
+			// get the data for what we care about and return
+			r.journal.RestartData()
+			if c > 0 {
+			fields:
```

Comment:
nit: maybe change to another name, or even better, put into a small function? Because `fields` is already taken an variable here, so it feels a little bit confusing.